### PR TITLE
feat: Pre-work needed for the lineage graph visualization

### PR DESCRIPTION
**Pull request by verdan**
### Summary of Changes

This PR moves the lineage implementation on the frontend out from the table metadata, and creates its own separate configuration and typings, along with its own state. This all will be needed to implement lineage graph visualization on a separate page without loading the complete table metadata every time. 

### Tests

All the required test cases have been added or updated where needed. 

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [ ] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [ ] PR includes a summary of changes.
- [ ] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does

### CODE REVIEW

**Comment by verdan** [comment]
The tests are failing because this requires the newer version of Amundsen common package. https://github.com/amundsen-io/amundsen/pull/1080

File: frontend/amundsen_application/static/.betterer.results [hunk 1]
```diff
@@ -563,6 +563,10 @@ exports[`eslint`] = {
     "js/ducks/lastIndexed/sagas.ts:1498244597": [
       [7, 2, 29, "\'action\' is defined but never used.", "566797395"]
     ],
+    "js/ducks/lineage/api/v0.ts:63450243": [
+      [19, 13, 26, "Expected the Promise rejection reason to be an Error.", "875716520"],
+      [44, 13, 26, "Expected the Promise rejection reason to be an Error.", "875716520"]
+    ],
```
Comment (Golodhros) [inline frontend/amundsen_application/static/.betterer.results:569]: These are new, any way of fixing them?

File: frontend/amundsen_application/static/js/ducks/lineage/index.spec.ts [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+    });
+
+    it('getTableLineage - returns the action to process success', () => {
+      const status = 200;
+      const action = getTableLineageSuccess(testLineage, status);
+      const { payload } = action;
+      expect(action.type).toBe(GetTableLineage.SUCCESS);
+      expect(payload.lineageTree).toBe(testLineage);
+      expect(payload.statusCode).toBe(status);
+    });
```
Comment (Golodhros) [inline frontend/amundsen_application/static/js/ducks/lineage/index.spec.ts:87]: Nit: I tend to group tests related to the same code by using a describe clause:
``
describe('getTableLineage', () => {
...
});
```

File: frontend/amundsen_application/static/js/ducks/lineage/reducer.ts [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+  columnName: string,
+  depth: number = 1,
+  direction: string = 'both'
+): GetColumnLineageRequest {
+  return {
+    type: GetColumnLineage.REQUEST,
+    payload: { key, depth, direction, columnName },
+    meta: {
+      analytics: {
+        name: `getColumnLineage`,
```
Comment (Golodhros) [inline frontend/amundsen_application/static/js/ducks/lineage/reducer.ts:78]: Let's namespace this one as `lineage/getColumnLineage`:
```suggestion
        name: `lineage/getColumnLineage`,
```

File: frontend/amundsen_application/static/js/ducks/lineage/reducer.ts [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+export function getTableColumnLineage(
+  key: string,
+  columnName: string
+): GetTableColumnLineageRequest {
+  return {
+    type: GetTableColumnLineage.REQUEST,
+    payload: { key, columnName },
+    meta: {
+      analytics: {
+        name: `getTableColumnLineage`,
```
Comment (Golodhros) [inline frontend/amundsen_application/static/js/ducks/lineage/reducer.ts:123]:
```suggestion
        name: `lineage/getTableColumnLineage`,
```

File: frontend/amundsen_application/static/js/pages/TableDetailPage/LineageLink/index.tsx [hunk 5]
```diff
@@ -16,7 +16,7 @@ const LineageLink: React.FC<LineageLinkProps> = ({
   tableData,
 }: LineageLinkProps) => {
   const config = AppConfig.tableLineage;
-  if (!config.isEnabled) return null;
+  if (!config.externalEnabled) return null;
```
Comment (Golodhros) [inline frontend/amundsen_application/static/js/pages/TableDetailPage/LineageLink/index.tsx:19]:
```suggestion
  if (!config.externalEnabled) {
      return null;
  }
```

File: frontend/amundsen_application/static/.betterer.results (hunk 1 above)
Comment (verdan) [reply frontend/amundsen_application/static/.betterer.results:569]: The results file already has a few such errors ignored. `Expected the Promise rejection reason to be an Error.`, so I am assuming this is a known issue.

Comment (Golodhros) [reply frontend/amundsen_application/static/.betterer.results:569]: I see...
We need to review this in detail, we might want to get rid of the eslint rule if we are OK with this.

**Review by Golodhros — APPROVED** [review]
LGTM JS-wise!

**Pull request merged**